### PR TITLE
Docs!

### PR DESCRIPTION
**Pull request by acpaquette**
Adds the initial doc files along with the necessary configs.

To create the docs:

- `cd /path/to/ale/ale/`
- `doxygen docs/config`
- `cd docs`
- `make html`

The resulting html will be located in the docs/build directory

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+# sys.path.insert(0, os.path.abspath('.'))
+import guzzle_sphinx_theme
+
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'ale'
+copyright = '2018, Adam Paquette'
+author = 'Adam Paquette'
```
Comment (jlaura) [inline docs/source/conf.py]: Can we swap this to be: `copyright = Public Domain author = USGS, Astrogeology Science Center`

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    #
+    # 'figure_align': 'htbp',
+}
+
+# Grouping the document tree into LaTeX files. List of tuples
+# (source start file, target name, title,
+#  author, documentclass [howto, manual, or own class]).
+latex_documents = [
+    (master_doc, 'ALE.tex', 'ALE Documentation',
+     'Adam Paquette', 'manual'),
```
Comment (jlaura) [inline docs/source/conf.py]: 'USGS, Astrogeology Science Center'

Comment (jlaura) [reply docs/source/conf.py]: or use the author variable

File: docs/source/conf.py (hunk 1 above)
Comment (acpaquette) [reply docs/source/conf.py]: :+1:

File: docs/source/conf.py (hunk 2 above)
Comment (acpaquette) [reply docs/source/conf.py]: :+1:

File: include/ale.h [hunk 3]
```diff
@@ -9,35 +9,112 @@
...
     linear,
+    /// Interpolate using spline interpolation
     spline
   };
 
-  nlohmann::json constructStateFromIsd(const std::string positionRotationData);
-
+  /**
+   *@brief Get the position of the spacecraft at a given time based on a set of coordinates, and their associated times
+   *@param coords A vector of double vectors of coordinates
```
Comment (Kelvinrr) [inline include/ale.h]: There has to be a way to document the type in a more explicit way. Although, let's get this PR merged in first and worry about it later.

**Pull request merged**